IANotificationsTestProject/In App Notifications/IANQueue.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+struct IANQueue<T>: Queuing {
+    private var array: [T] = []
+    
+    public init() {}
+    
+    var isEmpty: Bool {
+        array.isEmpty
+    }
+    
+    @discardableResult mutating func enqueue(_ element: T) -> Bool {
+        array.append(element)
+        return true
+    }
+    
+    mutating func dequeue() -> T? {
+        if !isEmpty {
+            return array.removeFirst()
+        } else {
+            return nil
+        }
```

Comment:
Since L19 has a `return`, you can drop the *else* in L20 (and remove one indentation level!)